src/ILCompiler.Compiler/src/CppCodeGen/CppWriter.cs

```diff
@@ -729,189 +729,194 @@ private String GetCodeForVirtualMethod(MethodDesc method, int slot)
             return sb.ToString();
         }
 
-        private void AppendVirtualSlots(CppGenerationBuffer sb, TypeDesc implType, TypeDesc declType)
+        private String GetCodeForNode(DependencyNode node, NodeFactory factory)
         {
-            var baseType = declType.BaseType;
-            if (baseType != null)
-                AppendVirtualSlots(sb, implType, baseType);
+            // virtual slots
+            var nodeData = (node as ObjectNode).GetData(factory, false);
```

Comment:
If `ObjectNode` is the only node type this method can handle, make the method take `ObjectNode` instead of `DependencyNode`?